chainer/functions/activation/randomized_leaky_relu.py

```diff
@@ -0,0 +1,106 @@
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+import numpy as np
+
+
+def _kern():
+    return cuda.elementwise(
+        'T cond, T x, T slope', 'T y',
+        'y = cond >= 0 ? x : (T)(slope * x)', 'rrelu')
+
+
+class RReLU(function.Function):
+    """Randomized Leaky rectifier unit."""
+
+    def __init__(self, lower=1. / 8, upper=1. / 3, train=True):
+        self.lower = lower
+        self.upper = upper
+        self.train = train
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+        type_check.expect(x_type.dtype.kind == 'f')
+
+    def forward_cpu(self, x):
+        y = x[0].copy()
+        if self.train:
+            self.r = np.random.uniform(self.lower, self.upper, x[0].shape[0:2])
```

Comment:
It seems the paper considers only the case when each input is one-dimensional, so I cannot say about two (or more) dimensional case from the paper. (Unfortunately, the paper does not look well written to me...)
I searched other RReLU implementation to inspect this, and I found that at least DL4J deals the random numbers are independent over all elements. https://github.com/deeplearning4j/nd4j/blob/master/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/activations/impl/ActivationRReLU.java